ldap/ldap-connector/ldap-connector-1.0.0/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/ldap/SearchEntry.java

```diff
@@ -0,0 +1,147 @@
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.connector.ldap;
+
+import javax.naming.NamingEnumeration;
+import javax.naming.NamingException;
+import javax.naming.directory.Attribute;
+import javax.naming.directory.Attributes;
+import javax.naming.directory.DirContext;
+import javax.naming.directory.SearchControls;
+import javax.naming.directory.SearchResult;
+
+import org.apache.axiom.om.OMAbstractFactory;
+import org.apache.axiom.om.OMElement;
+import org.apache.axiom.om.OMFactory;
+import org.apache.axiom.om.OMNamespace;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.SynapseException;
+import org.apache.synapse.core.axis2.Axis2MessageContext;
+import org.apache.synapse.transport.nhttp.NhttpConstants;
+import org.wso2.carbon.connector.core.AbstractConnector;
+import org.wso2.carbon.connector.core.ConnectException;
+import org.wso2.carbon.connector.core.util.ConnectorUtils;
+
+public class SearchEntry extends AbstractConnector {
+    public static final String OBJECT_CLASS = "objectClass";
+    public static final String FILTERS = "filters";
+    public static final String DN = "dn";
+    public static final String ATTRIBUTES = "attributes";
+    protected static Log log = LogFactory.getLog(SearchEntry.class);
+    @Override
+    public void connect(MessageContext messageContext) throws ConnectException {
+        String objectClass = (String)getParameter(messageContext, OBJECT_CLASS);
+        String filter  = (String)getParameter(messageContext, FILTERS);
+        String dn = (String)getParameter(messageContext, DN);
+        String returnAttributes[] = ((String)getParameter(messageContext, ATTRIBUTES)).split(",");
+
+        OMFactory factory = OMAbstractFactory.getOMFactory();
+        OMNamespace ns = factory.createOMNamespace(LDAPConstants.CONNECTOR_NAMESPACE, "ns");
+        OMElement result = factory.createOMElement("result", ns);
+
+        try {
+            DirContext context = LDAPUtils.getDirectoryContext(messageContext);
+
+            String attrFilter = generateAttrFilter(filter);
+            String searchFilter = generateSearchFilter(objectClass, attrFilter);
+            NamingEnumeration<SearchResult> results = null;
+            try {
+                results = searchInUserBase(dn, searchFilter, returnAttributes, SearchControls.SUBTREE_SCOPE, context);
+
+                SearchResult entityResult = null;
+
```

Comment:
Empty line is not needed here. Please consider this in whole PR. You can use the empty line to separate bulky code to make it understandable.